lit_gpt/lora.py

```diff
@@ -335,30 +323,34 @@ def zero_pad(self, x: torch.Tensor) -> torch.Tensor:
         result = result.index_copy(1, self.lora_ind, x.reshape(-1, shape))  # (4096, 256)
         return result.view((*x.shape[:-1], self.out_features)).transpose(0, 1)  # (64, 64, 384)
 
-    def merge(self):
-        """Merges the LoRA weights into the full-rank weights (W = W + delta_W)."""
+    def merge(self, verbose = True):
```

Comment:
This PR combines a few too many changes into one IMO. Could we perhaps break it into two parts? 1) the removal of the merge_weights arguments and 2) the validation logic. It would be easier to review each part alone and faster to land.

Reply:
I think you are right. Sometimes I get too carried away. Sorry for that.

> 2) the validation logic

What do you mean by that? Printing if the weights cannot be merged?

Reply:
The validation logic for r > 0 and so on.